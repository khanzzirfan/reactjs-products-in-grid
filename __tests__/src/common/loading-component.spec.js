import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LoadingComponent from 'src/common/loading-component';

describe("loading component test", () => {

    let wrapper = null;
    beforeEach(() => {
        wrapper = shallow(<LoadingComponent />);
    })

    it("component should be defined", () => {
        expect(wrapper).toBeDefined();
    });

    it("component should match to snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("it should render with correct class names", ()=>{
        expect(wrapper.find('.loading').exists()).toBeTruthy();
        expect(wrapper.find('.loading-bar').exists()).toBeTruthy();
        expect(wrapper.find('.p-2').exists()).toBeTruthy();

        expect(wrapper.find('.text-center').exists()).toBeTruthy();
        expect(wrapper.find('.flex-column').exists()).toBeTruthy();
        expect(wrapper.find('.justify-content-center').exists()).toBeTruthy();
        
        expect(wrapper.find('.loading-bar').length).toBe(4);
    });

    it("it should render with correct text", ()=>{
        expect(wrapper.text()).toContain("Loading....");
        
    });

});