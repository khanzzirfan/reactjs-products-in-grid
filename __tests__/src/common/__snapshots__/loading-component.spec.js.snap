// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading component test component should match to snapshot 1`] = `
<div
  className="col-md-12 text-center"
>
  <div
    className="d-flex flex-column justify-content-center"
  >
    <div
      className="p-2"
    >
      <div
        className="loading"
      >
        <div
          className="loading-bar"
        />
        <div
          className="loading-bar"
        />
        <div
          className="loading-bar"
        />
        <div
          className="loading-bar"
        />
      </div>
    </div>
    <div
      className="p-2"
    >
       
    </div>
    <div
      className="p-2"
    >
       
    </div>
    <div
      className="p-2"
    >
       Loading....
    </div>
  </div>
</div>
`;
